src/java/main/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -915,11 +915,13 @@ protected void pRequest(Request request) throws RequestProcessorException {
     private List<ACL> removeDuplicates(List<ACL> acl) {
 
         ArrayList<ACL> retval = new ArrayList<ACL>();
-        Iterator<ACL> it = acl.iterator();
-        while (it.hasNext()) {
-            ACL a = it.next();
-            if (retval.contains(a) == false) {
-                retval.add(a);
+        if(acl != null) {
```

Comment:
Good point. I added the check for nulls on `removeDuplicates` because `Set` doesn't accept null entries. So, if there is a null in `acl` list then the first snippet I suggested would throw an exception.